src/main/java/be/uantwerpen/minelabs/renderer/BohrBlueprintEntityRenderer.java

```diff
@@ -82,7 +203,7 @@ private List<Integer> getElectronShellConfiguration(int nE) {
         // TODO: placeholder that fills until max. Should use energy based filling of shells.
         // TODO: if too slow, place cache over this function
         List<Integer> result = new ArrayList<>(7);
-        for (int c : ELECTRON_SHELL_CAPACITIES) {
+        for (int c : ELECTRON_SHELL_MAPPINGS.get(nE)) {
```

Comment:
This loop is no longer needed as you have the exact distribution stored already. Just return the appropriate one directly and change the return type. Might need to change how this function is used as well.

Reply:
Removed loop, return type is int[] now.

Reply:
Perfect! Can you remove the profiler and TODOs, then it's ready for the next PR.